### PR TITLE
Build for Scala.js 1.0.0-RC1

### PR DESCRIPTION
**Pull request by xerial**

### CODE REVIEW

**Comment by julienrf** [comment]
It looks like we should disable mima when building for Scala.js 1.0.0-RC1

**Comment by xerial** [comment]
I don't know why, but Travis CI environment is broken.

**Comment by lrytz** [comment]
cleared the travis caches and re-started the builds (cause: same as https://github.com/scala/scala/pull/8528), the results should be acurate now.

File: build.sbt [hunk 1]
```diff
@@ -14,7 +14,13 @@ lazy val commonSettings = Seq(
                                                  |See the NOTICE file distributed with this work for
                                                  |additional information regarding copyright ownership.
                                                  |""".stripMargin)),
-  scalaModuleMimaPreviousVersion := Some("2.1.1"),
+  scalaModuleMimaPreviousVersion := {
+    // We need to create a release version first to use MiMa
+    if(sys.env.get("SCALAJS_VERSION").exists(_.startsWith("1.0.0-M")))
```
Comment (julienrf) [inline build.sbt]: Should be `1.0.0-`, no? (without the `M`)

Comment (ekrich) [reply build.sbt]: I was thinking that as well if we are changing to `RC1` ...

Comment (julienrf) [reply build.sbt]: Also, it should be `forall` rather than `exists` because we still want to enable MiMa when building JVM artifacts.

**Comment by xerial** [comment]
Applied the suggested fixes. At least, compatXXXJs/tests worked in my machine.

File: build.sbt (hunk 1 above)
Comment (julienrf) [reply build.sbt]: Uh, I’m sorry but my reasoning was wrong, using `exists` as you initially did was the correct thing to do…

**Comment by julienrf** [comment]
Could you please sign the Scala CLA?
https://www.lightbend.com/contribute/cla/scala

**Comment by xerial** [comment]
I think I have already signed to that, but let me double check.

**Comment by xerial** [comment]
Signing is done.

**Comment by julienrf** [comment]
Thanks @xerial! It seems that something is wrong with the CLA system… Does anyone have an idea of what we can do? (@lrytz or @SethTisue?)

**Comment by lrytz** [comment]
[`checkCLA.sh`](https://github.com/scala/scala-collection-compat/blob/master/admin/checkCLA.sh#L7) does

```
curl -u dummy4dummy:dummy2dummy -s "https://api.github.com/repos/scala/scala-collection-compat/pulls/268"
```

which gives me, on my machine

```
{
  "message": "Bad credentials",
  "documentation_url": "https://developer.github.com/v3"
}
```

If I remove the dummy login, it works

```
$> curl -s "https://api.github.com/repos/scala/scala-collection-compat/pulls/268"
{
  "url": "https://api.github.com/repos/scala/scala-collection-compat/pulls/268",
  "id": 344705263,
  "node_id": "MDExOlB1bGxSZXF1ZXN0MzQ0NzA1MjYz",
  "html_url": "https://github.com/scala/scala-collection-compat/pull/268",
  "diff_url": "https://github.com/scala/scala-collection-compat/pull/268.diff",
  "patch_url": "https://github.com/scala/scala-collection-compat/pull/268.patch",
  "issue_url": "https://api.github.com/repos/scala/scala-collection-compat/issues/268",
...
```

I saw the same in some other repo while ago, don't remember which.. removed the login.

**Comment by lrytz** [comment]
hmm, scratch that, i had an old checkout locally; i did remove the dummy login in this repo https://github.com/scala/scala-collection-compat/commit/78ce8686de4fe3b8213cba98ad896d895b31d4b2.

**Comment by lrytz** [comment]
did a few re-runs, it's green now. not sure what's behind the cla check failure..

**Comment by julienrf** [comment]
Thank you all for the team work ;)

**Pull request merged**